### ISSUE TITLE
[HOLD for payment 2023-07-20] [HOLD for payment 2023-07-17] [$1000] Chat - Area outside the group avatar is clickable

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#22445]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Issue found when executing PR** https://github.com/Expensify/App/pull/21402
## Action Performed:
1. Go to staging.new.expensify.com
2. Create a group of five
3. Resize the conversation view to small screen
4. Tap outside the group avatar

## Expected Result:
The area outside the group avatar is not clickable

## Actual Result:
The area outside the group avatar is clickable

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.38.3

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/eb74761a-e16e-4241-ac8f-c7c3da50c9e5

![Bug6120101_bandicam_2023-07-08_04-10-54-161](https://github.com/Expensify/App/assets/78819774/ba4212e3-51d5-4675-80f4-bf3439eb8d8e)

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016720a203167c26a6</li>
        <li>Upwork Job ID: 1678527009111691264</li>
        <li>Last Price Increase: 2023-07-10</li>
    </ul>
</details>

**Comment by DrLoopFall** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The area outside the group avatar is clickable

### What is the root cause of that problem?
There are actually two problems here,
1. We are using padding here, which expands the clickable area
![image](https://github.com/Expensify/App/assets/120749263/04076cd4-683e-440c-82ad-54efea559483)
https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/pages/home/report/ReportActionItemCreated.js#L76
2. We are using `left` for positioning/shifting avatars but the `PressableWithoutFeedback` itself is not shifted, which causes it to be clickable in both the original and shifted positions.
![image](https://github.com/Expensify/App/assets/120749263/1217d86e-8d39-43ed-86a0-cf52381957f1)
https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/styles/StyleUtils.js#L906

### What changes do you think we should make in order to solve the problem?
Make individual avatars/row clickable.

Remove pressable from here
https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/pages/home/report/ReportActionItemCreated.js#L74

Wrap this with `PressableWithoutFeedback`
https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/components/MultipleAvatars.js#L157
or this
https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/components/MultipleAvatars.js#L152

We add `onPress` prop in `MultipleAvatars`

### What alternative solutions did you explore? (Optional)
Replace paddings with margins.
```js
- style={[styles.ph5, styles.pb3, styles.alignSelfStart]} 
+ style={[styles.mh5, styles.mb3, styles.alignSelfStart]} 
```
Use `margin-left` instead of `left` for positioning the avatar.
```js
- left: -(overlapSize * index)
+ marginLeft: index ? -overlapSize : 0
```
We also need to update(or remove) the width
https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/components/MultipleAvatars.js#L138-L144

NOTE: In **alternate solution** this gap is still clickable.
![image](https://github.com/Expensify/App/assets/120749263/a1bcd3bf-392c-431e-b67c-b95b10608196)

**Comment by situchan** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Chat - welcome area outside the group avatar is clickable

### What is the root cause of that problem?
There are 2 root causes.
- red space comes from [wrong style](https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/pages/home/report/ReportActionItemCreated.js#L76) in PressableWithoutFeedback of `ReportActionItemCreated`
- green right space comes from [wrong calculation](https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/components/MultipleAvatars.js#L134-L145) of `avatarContainerStyles` width. It was old logic to treat count of avatars as 4 if more than 4 but after #15929, they're wrapped into 2 lines so count is now `Math.ceil(icon length / 2)`. In above example, count is 3.

<img width="373" alt="Screenshot 2023-07-10 at 7 28 44 AM" src="https://github.com/Expensify/App/assets/108292595/d9e2cda3-9847-4d5b-a308-3d0bb8b0e70d">


### What changes do you think we should make in order to solve the problem?
Based on above analysis,
- replace padding with margin
https://github.com/Expensify/App/blob/60f48175c9e3342ef192d878218820e5ed5a76cd/src/pages/home/report/ReportActionItemCreated.js#L76

After fix:
<img width="373" alt="Screenshot 2023-07-10 at 7 29 16 AM" src="https://github.com/Expensify/App/assets/108292595/df2cb201-cf5b-4356-b75c-ee9f3ab9560b">

- refactor width calculation logic: get max length of row
```
        const length = avatarRows.length > 1 ? Math.max(avatarRows[0].length, avatarRows[1].length) : avatarRows[0].length;
        const width = oneAvatarSize.width + overlapSize * 2 * (length - 1) + oneAvatarBorderWidth * (length * 2);
```

After fix:
<img width="373" alt="Screenshot 2023-07-10 at 7 30 03 AM" src="https://github.com/Expensify/App/assets/108292595/d999553f-441b-452f-977c-0f9c9cc8fe74">

**Comment by stitesExpensify** [comment]
Okay so everyone so far has pointed to [this line](https://github.com/allroundexperts/Expensify/blame/main/src/pages/home/report/ReportActionItemCreated.js#L76) as the culprit, but as far as I can tell that hasn't been updated in 6 months. What happened in this staging deploy that caused things to break since it's working properly on production?

**Comment by stitesExpensify** [comment]
Was it just something that was missed by this pr maybe? https://github.com/Expensify/App/pull/20512/files

**Comment by situchan** [comment]
@stitesExpensify it was missed by https://github.com/Expensify/App/issues/15929

**Comment by situchan** [comment]
In my proposal, first issue existed long time. 2nd issue was caused by https://github.com/Expensify/App/issues/15929

**Comment by DrLoopFall** [comment]
@stitesExpensify I think the `width` was set as a workaround for the issue described in my [proposal](https://github.com/Expensify/App/issues/22445#issuecomment-1627413019) (the 2nd issue)

> We are using `left` for positioning/shifting avatars but the `PressableWithoutFeedback` itself is not shifted, which causes it to be clickable in both the original and shifted positions.

If we use `margin-left` instead, we can remove the `width`, which may prevent any future issues.

**Comment by stitesExpensify** [comment]
Hm okay. I think that your proposal makes sense @situchan. @puneetlath @WikusKriek what do you think since it looks like this PR is what caused the issue

**Comment by puneetlath** [comment]
@Christinadobrzyn FYI, I did the fix since it was a deploy blocker. 

But I used @situchan's solution, so let's make sure they get paid out. That's why I added the external label. Thanks!

**Comment by Christinadobrzyn** [comment]
Invited @situchan to the Upwork job 

Internal job: https://www.upwork.com/ab/applicants/1678527009111691264/job-details
External job posting: https://www.upwork.com/ab/applicants/1678527009111691264/job-details

Let me know if there's anyone else to pay!

**Comment by elghanbibadr** [comment]
PROPOSAL

**Comment by elghanbibadr** [comment]
Contributor details
Your Expensify account email: badrghanbi50@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~015c5fa1a95af5c54d

**Comment by elghanbibadr** [comment]
Contributor details
Your Expensify account email: bghanbi50@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~015c5fa1a95af5c54d

**Comment by elghanbibadr** [comment]
> Invited @situchan to the Upwork job
> 
> Internal job: https://www.upwork.com/ab/applicants/1678527009111691264/job-details External job posting: https://www.upwork.com/ab/applicants/1678527009111691264/job-details
> 
> Let me know if there's anyone else to pay!

**Comment by elghanbibadr** [comment]
would I get hired ?

Le mar. 11 juil. 2023 à 00:23, melvin-bot[bot] ***@***.***> a
écrit :

**Comment by DrLoopFall** [comment]
Hi @stitesExpensify, @puneetlath and @Christinadobrzyn,
May I know why @situchan's [proposal](https://github.com/Expensify/App/issues/22445#issuecomment-1628355796) was selected over [my proposal](https://github.com/Expensify/App/issues/22445#issuecomment-1627413019), as my proposal fixes the actual root cause of the issue, not fixing some [workaround](https://github.com/Expensify/App/issues/22445#issuecomment-1629454284).

Also, @situchan's proposal is almost a duplicate of [my proposal](https://github.com/Expensify/App/issues/22445#issuecomment-1627413019).

**Comment by situchan** [comment]
@DrLoopFall please check [PR](https://github.com/Expensify/App/pull/22595). It was already merged.
Calculating container width correctly is already existing logic and it's not a workaround, and not dupe of your proposal.
If you think this logic should be refactored, please feel free to raise discussion on slack.

**Comment by DrLoopFall** [comment]
@situchan, I didn't say your proposed login is a workaround, I said the solution you provided was fixing a workaround that was already there. 

> @stitesExpensify I think the `width` was set as a workaround for the issue described in my [proposal](https://github.com/Expensify/App/issues/22445#issuecomment-1627413019) (the 2nd issue)
> 
> > We are using `left` for positioning/shifting avatars but the `PressableWithoutFeedback` itself is not shifted, which causes it to be clickable in both the original and shifted positions.
> 
> If we use `margin-left` instead, we can remove the `width`, which may prevent any future issues.

**Comment by puneetlath** [comment]
@DrLoopFall I didn't totally follow your proposal and since it was a deploy blocker, I went with @situchan's since I understood it. It's definitely possible that your solution is better. 

It looks like the solution we implemented created a bug. Perhaps you can post your proposal on this issue and we can discuss further? https://github.com/Expensify/App/issues/22632

**Comment by DrLoopFall** [comment]
@puneetlath Thank you,
I've submitted a [proposal](https://github.com/Expensify/App/issues/22632#issuecomment-1631418875) there, as I already said using `left` was causing the problem, please have a look [there](https://github.com/Expensify/App/issues/22632#issuecomment-1631418875).

**Comment by sonialiap** [comment]
Paid ✔️
